dlog/commitment/src/commitment.rs

```diff
@@ -196,20 +203,27 @@ impl<G: AffineCurve> SRS<G> {
         let p = {
             let mut p = DensePolynomial::<Fr<G>>::zero();
             let mut scale = Fr::<G>::one();
+
```

Comment:
Can you declare a function above this code like

```rust
// TODO: Figure out how to return iterator here
fn segments<F:Field>(n: usize, p : DensePolynomial<F>, degree_bound: Option<usize>) -> Vec<DensePolynomial<F>> {
    let d = p.coeffs.len();
    let segments = (d + (n - 1)) / n;

    let mut res : Vec<_> =
        (0..segments).map(|i| {
            DensePolynomial::from_coefficients_slice(
                &p.coeffs[(i*n)..((i+1)*n)])
        }).collect() ;

    match degree_bound {
        Some(bound) => res.push(
            DensePolynomial::from_coefficients_slice(
                &p.coeffs[(segments - 1)*n..]).shiftr(n - (bound % n))),
        None => ()
    };

    res
}
```

and then in here just do

```rust
            for (p_i, degree_bound) in plnms.iter() {
                for seg in segments(self.g.len(), p_i, *degree_bound) {
                    p += &seg.scale(scale);
                    scale *= &polyscale;
                }
            }
```

Reply:
This doesn't quite seem to work (unless I misunderstand):

1. p.coeffs[(i*n)..((i+1)*n)]) could result in boundary access violation
2. This function puts the shifted segment into the same vector/iterator. However it has to be handled separately since it has to be mixed into the accumulator poly shifted to the rigjt edge of SRS..